test-specifications/token-service/tokenCreateTransaction.md

```diff
@@ -0,0 +1,1154 @@
+| 4       | Creates a token with a valid ECDSAsecp256k1 private key as its admin key                                         | name="testname", symbol="testsymbol", treasuryAccountId=<OPERATOR_ACCOUNT_ID>, adminKey=<VALID_ECDSA_SECP256K1_PRIVATE_KEY>, commonTransactionParams.signers=[<VALID_ECDSA_SECP256K1_PRIVATE_KEY>]      | The token creation succeeds and the token has the new ECDSAsecp256k1 private key as its admin key. | N                 |
+| 5       | Creates a token with a valid KeyList of ED25519 and ECDSAsecp256k1 private and public keys as its admin key      | name="testname", symbol="testsymbol", treasuryAccountId=<OPERATOR_ACCOUNT_ID>, adminKey=<VALID_KEYLIST>, commonTransactionParams.signers=[<KEYS_IN_KEYLIST>]                                            | The token creation succeeds and the token has the new KeyList as its admin key.                    | N                 |
+| 6       | Creates a token with a valid KeyList of nested Keylists (three levels) as its admin key                          | name="testname", symbol="testsymbol", treasuryAccountId=<OPERATOR_ACCOUNT_ID>, adminKey=<VALID_NESTED_KEYLIST>, commonTransactionParams.signers=[<KEYS_IN_NESTED_KEYLIST>]                              | The token creation succeeds and the token has the new nested KeyList as its admin key.             | N                 |
+| 7       | Creates a token with a valid ThresholdKey of ED25519 and ECDSAsecp256k1 private and public keys as its admin key | name="testname", symbol="testsymbol", treasuryAccountId=<OPERATOR_ACCOUNT_ID>, adminKey=<VALID_THRESHOLD_KEY>, commonTransactionParams.signers=[<KEYS_IN_THRESHOLD_KEY>]                                | The token creation succeeds and the token has the new ThresholdKey as its admin key.               | N                 |
+| 8       | Creates a token with a valid key as its admin key but doesn't sign with it                                       | name="testname", symbol="testsymbol", treasuryAccountId=<OPERATOR_ACCOUNT_ID>, adminKey=<VALID_KEY>                                                                                                     | The token creation fails with an INVALID_SIGNATURE response code from the network.                 | N                 |
+| 9       | Creates a token with an invalid key as its admin key                                                             | name="testname", symbol="testsymbol", treasuryAccountId=<OPERATOR_ACCOUNT_ID>, adminKey=<INVALID_KEY>                                                                                                   | The token creation fails with an SDK internal error.                                               | N                 |
```

Comment:
I think we do validate key inputs? Just looking through a couple of the SDKs now, looks like most at least check length, but then uses internal libraries to create some internal key representation and catches bad keys. Otherwise what would be the use of `BadKeyException`?